core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java

```diff
@@ -293,7 +293,10 @@ private void selectPrimaryGWFromGatewayInstances(Collection<InstanceGroup> insta
                 .findFirst();
 
         if (gwInstanceGroup.isPresent()) {
-            List<InstanceMetaData> gwInstances = instanceMetaDataService.findAllByInstanceGroupAndInstanceStatus(gwInstanceGroup.get(), InstanceStatus.CREATED);
+            // Ordered list of metadata to guarantee consistent primary gateway generation across multiple cluster recoveries
```

Comment:
If we had primary GW then we select it here: https://github.com/hortonworks/cloudbreak/blob/master/core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java#L185-L189
So in upgrade (repair) we restore the PGW to the node with the same FQDN. I don't understand why this change needed?